### PR TITLE
PIM-6033: Fix validation issue when you add an empty attribute option

### PR DESCRIPTION
**Pull request by anaelChardan**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md) 

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |  :negative_squared_cross_mark:
| Added Behats                      | :white_check_mark:
| Changelog updated                 | :white_check_mark:
| Review and 2 GTM                  | :clock1:
| Micro Demo to the PO (Story only) | :clock1:
| Migration script                  | :negative_squared_cross_mark:
| Tech Doc                          |  :negative_squared_cross_mark:

:white_check_mark: Done and pass

:red_circle: Done but fail

:clock1: Done but pending

:negative_squared_cross_mark: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js [hunk 1]
```diff
@@ -411,8 +410,11 @@ define(
                         this.updateEditionStatus();
                     }
                 }
-
-                this.currentlyEditedItemView = attributeOptionRow;
+                if (attributeOptionRow.model.id)
+                {
+                    this.currentlyEditedItemView = attributeOptionRow;
+                }
+                // this.currentlyEditedItemView = attributeOptionRow;
```
Comment (a2xchip) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js]: Is it WIP or we do not need this comment, same as these on line 387 and 365?

**Comment by a2xchip** [comment]
PR title ``` you add an and ```

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js (hunk 1 above)
Comment (anaelChardan) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js]: It's a WIP :)

Comment (anaelChardan) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js]: Comments on line 387 and 365 were here before this ticket and are here to explain the action :-). I fixed your comment on line 417 :-)

Comment (a2xchip) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/pim-attributeoptionview.js]: Nice!
I see they were already present there, sometimes i find ```console.logs``` in js, does not mean it couldn't be removed during clean up ;-)

File: CHANGELOG-1.6.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# 1.6.x
+
+## Bug fixes
+
+- PIM-6033: Fix validation issue when you add an attribute option line blank
```
Comment (ahocquard) [inline CHANGELOG-1.6.md]: "a blank attribute option line" , no ?

File: features/Context/Page/Attribute/Creation.php [hunk 3]
```diff
@@ -58,22 +59,58 @@ public function findField($name)
...
     {
         if (null === $this->getElement('attribute_option_table')->find('css', '.attribute_option_code')) {
+            $this->addAnOption();
+        }
+
+        $this->fillLastOption($name, $labels);
+        $this->saveLastOption();
+    }
+
+    public function addAnOption()
```
Comment (ahocquard) [inline features/Context/Page/Attribute/Creation.php]: it seems weird and redundant to have addAnOption and addOption functions.

File: CHANGELOG-1.6.md (hunk 2 above)
Comment (anaelChardan) [inline CHANGELOG-1.6.md]: It was the ticket name :-D

File: features/Context/Page/Attribute/Creation.php (hunk 3 above)
Comment (anaelChardan) [inline features/Context/Page/Attribute/Creation.php]: the name of addOption is very not clear, in fact addOption : add / fill / save an option but addAnOption only click on the button addAnOption :-)

File: CHANGELOG-1.6.md (hunk 2 above)
Comment (anaelChardan) [inline CHANGELOG-1.6.md]: But I can correct

Comment (ahocquard) [inline CHANGELOG-1.6.md]: better imo, you can change the ticket name if it sounds better

File: features/Context/Page/Attribute/Creation.php (hunk 3 above)
Comment (ahocquard) [inline features/Context/Page/Attribute/Creation.php]: what about createOption and addOption ?

Comment (anaelChardan) [inline features/Context/Page/Attribute/Creation.php]: Why not ;-)

File: features/Context/Page/Attribute/Creation.php [hunk 4]
```diff
@@ -58,22 +59,58 @@ public function findField($name)
...
     {
         if (null === $this->getElement('attribute_option_table')->find('css', '.attribute_option_code')) {
+            $this->createOption();
+        }
+
+        $this->fillLastOption($name, $labels);
+        $this->saveLastOption();
+    }
+
+    public function createOption()
```
Comment (anaelChardan) [inline features/Context/Page/Attribute/Creation.php]: I agree with you, the original method wasn't named correctly, and as it's a public method and in a support ticket I can't rename the original method :-S ^^

File: features/Context/Page/Attribute/Creation.php [hunk 5]
```diff
@@ -58,22 +59,58 @@ public function findField($name)
...
         foreach ($labels as $locale => $label) {
             $row->find('css', sprintf('.attribute-option-value[data-locale="%s"]', $locale))->setValue($label);
         }
-        $row->find('css', '.btn.update-row')->click();
+    }
+
+    public function saveLastOption()
+    {
+        $this->spin(function () {
+            $this->getLastOption()->find('css', '.btn.update-row')->click();
```
Comment (jjanvier) [inline features/Context/Page/Attribute/Creation.php]: I don't understand why we spin the click?
why isn't it outside the spin?

Comment (anaelChardan) [inline features/Context/Page/Attribute/Creation.php]: You right !

**Pull request merged**